astropy/coordinates/matrix_utilities.py

```diff
@@ -4,9 +4,12 @@
 Utililies used for constructing and inspecting rotation matrices.
 """
 
+__all__ = ["is_rotation_or_reflection", "rotation_matrix"]
```

Comment:
I see that `matrix_transpose` has not been included here.  Is it considered public or private?

Reply:
With numpy 2.0 having introduced `.mT` as well as `np.matrix_transpose`, `matrix_transpose` will be deprecated once 2.0 is the minimum version. Given that, perhaps we should not expose our own version? Note that the next version of astropy will drop support for numpy < 2.0 (at least, https://scientific-python.org/specs/spec-0000/ suggests that we should drop support for 1.26 third quarter of 2025).